--- am-utils-6.1.5/m4/macros/linux_headers.m4.kzak	2005-02-19 22:38:58.000000000 +0100
+++ am-utils-6.1.5/m4/macros/linux_headers.m4	2006-12-18 21:28:04.000000000 +0100
@@ -14,7 +14,14 @@
 ],
 [
 if (argc > 1)
+#ifdef UTS_RELEASE
   printf("%s", UTS_RELEASE);
+#else
+# define AMU_MA(a) ((a) >> 16)
+# define AMU_MI(a) (((a) & 0xffff) >> 8)
+# define AMU_PL(a) ((a) & 0xff)
+ printf("%d.%d.%d", AMU_MA(LINUX_VERSION_CODE), AMU_MI(LINUX_VERSION_CODE), AMU_PL(LINUX_VERSION_CODE));
+#endif
 ],
 [ host_header_version=$value ],
 [ echo
